camel-idea-plugin/src/main/java/org/apache/camel/idea/annotator/CamelSimpleAnnotator.java

```diff
@@ -49,7 +50,7 @@ boolean isEnabled() {
      */
     void validateText(@NotNull PsiElement element, @NotNull AnnotationHolder holder, @NotNull String text) {
         boolean hasSimple = text.contains("${") || text.contains("$simple{");
-        if (hasSimple && CamelIdeaUtils.isCamelSimpleExpression(element)) {
+        if (hasSimple || CamelIdeaUtils.isCamelSimpleExpression(element)) {
```

Comment:
I previously removed this and you made comment, that it would not work for expression like this
`.log("Result from query ${body}")`

Is the CamelIdeaUtils.isCamelSimpleExpression(element) handling this case also?